torch/nn/modules/module.py

```diff
@@ -331,14 +334,16 @@ def remove_from(*dicts):
                     raise AttributeError(
                         "cannot assign module before Module.__init__() call")
                 remove_from(self.__dict__, self._parameters, self._buffers)
-                value._name = name
+                if value is not None:
+                    value._name = name
                 modules[name] = value
             elif modules is not None and name in modules:
                 if value is not None:
                     raise TypeError("cannot assign '{}' as child module '{}' "
                                     "(torch.nn.Module or None expected)"
                                     .format(torch.typename(value), name))
-                value._name = name
+                if value is not None:
+                    value._name = name
```

Comment:
nit: I just saw that there is a `raise TypeError` just above, `if value is not None`. We can then maybe remove this condition + assignment altogether?

Reply:
Of course, thanks @fmassa